python/triton/compiler/code_generator.py

```diff
@@ -1267,6 +1267,9 @@ def visit_Call(self, node):
             if '_generator' in sig.parameters:
                 extra_kwargs['_generator'] = self
             try:
+                # Copy tuples to prevent builtins from modifying in place
+                args = [x.copy() if isinstance(x, language.tuple) else x for x in args]
+                kws = {k: v.copy() if isinstance(v, language.tuple) else v for k, v in kws.items()}
```

Comment:
Can you change `make_tensor_descriptor` to copy instead of changing this in the frontend. I'm not sure this is the direction we want to go.

Reply:
Good news! A quick audit of `semantic.py` doesn't show any other mutations!